src/python_framework_v02/troute/nhd_io.py

```diff
@@ -303,80 +345,139 @@ def get_ql_from_wrf_hydro_mf(
 def drop_all_coords(ds):
     return ds.reset_coords(drop=True)
 
+def write_to_netcdf(f, variables, datatype = 'f4'):
+    
+    '''
+    Quickly append or overwrite variable data in NetCDF files by leveraging the netCDF4 library. 
+    For additional documentation on netCDF4: https://unidata.github.io/netcdf4-python/#version-157
+    
+    Arguments:
+    ----------
+    f (Path): Name of netCDF file to hold dataset. Can also be a python 3 pathlib instance
+    variables (dict): dictionary keys are variable names (strings), dictionary values are tuples:
+                           (
+                               variable data (numpy array - 1D must be same size as variable dimension), 
+                               variable dimension name (string) that already exist in netCDF file, 
+                               variable attributes (dict, keys are attribute names and values are attribute contents),
+                           )
+    datatype: numpy datatype object, or a string that describes a numpy dtype object.
+              Supported specifiers include: 'S1' or 'c' (NC_CHAR), 'i1' or 'b' or 'B' (NC_BYTE),
+              'u1' (NC_UBYTE), 'i2' or 'h' or 's' (NC_SHORT), 'u2' (NC_USHORT), 'i4' or 'i' or 'l' (NC_INT),
+              'u4' (NC_UINT), 'i8' (NC_INT64), 'u8' (NC_UINT64), 'f4' or 'f' (NC_FLOAT), 'f8' or 'd' (NC_DOUBLE)
+    
+    NOTES:
+    - the netCDF files we want to append/edit must have write permission!
+    '''
+    
+    with netCDF4.Dataset(
+        filename = f,
+        mode = 'r+',
+        format = "NETCDF4"
+    ) as ds:
+
+        for varname, (vardata, dim, attrs) in variables.items():
+            
+            # check that dimension exists
+            if dim not in list(ds.dimensions.keys()):
+                LOG.error("The dimensions %s could not be found in file %s" % (dim, f))
+                LOG.error("Aborting writing process for %s. No data were written to this file" % f)
+                return        
+            
+            # check that dimension size and variable data size agree
+            dim_size = ds.dimensions[dim].size
+            if vardata.size != dim_size:
+                LOG.error("Cannot write data of size %d to variable with dimension size of %d" % (vardata.size, dim_size))
+                LOG.error("Aborting writing process for %s. No data were written to this file" % f)
+                return
+            
+            # check that varname doesn't already exist
+            # if it does, then overwrite it
+            if varname in list(ds.variables.keys()):
+
+                ds[varname][:] = vardata
 
-def write_q_to_wrf_hydro(
+            # if variable does not exist, create new one
+            else:
+
+                # create a new variable
+                y = ds.createVariable(
+                    varname = varname,
+                    datatype = datatype,
+                    dimensions = (dim,),
+                    fill_value = np.nan
+                )
```

Comment:
If a t-route variables do not already exist in either CHRTOUT or HYDRO_RST files, then create new variables and fill them with t-route data.